### ISSUE TITLE
Use travis matrix to test more systems

### ISSUE THREAD

**Issue by jubalh** [profanity-im/profanity#1100]
We should test more systems (which are up to date enough to have our dependencies).

- [ ] Arch Linux
- [ ] Debian Testing
- [x] openSUSE Tumbleed
- [ ] Mac OS X

We use Docker so we dont limit ourselves to the distributions that travis supports. I'm not sure whether that will create problems for us when we try to test OS X.

We will need separate docker files to init each system, update the travis.yml to use a build matrix and call the common travis-build.sh.

**Comment by paulfariello** [comment]
OS X is supported by travis.

**Comment by jubalh** [comment]
But I'm not sure if you can mix using docker with what they natively support.

**Comment by optmzr** [comment]
@jubalh Could I give this a try?

**Comment by jubalh** [comment]
@optmzr Sure!

AFAIK if you create a PR with the changed travis files it should then use those files for that PR so we can test it.
And maybe this helps: https://media.ccc.de/v/1354-continuous-integration-using-travis-and-docker/

Snapper uses docker and this [travis.yml](https://github.com/openSUSE/snapper/blob/master/.travis.yml) file shows how they do it.
In their case they have various dockerfiles, like [Dockerfile.Debian](https://github.com/openSUSE/snapper/blob/master/Dockerfile.debian) etc.
So basically we need the same to install the different OSes and then call `./travis-build.sh` in each of those.
